### ISSUE TITLE
Trying to have incline on my proform TDF [BUG]

### ISSUE THREAD

**Issue by mminville** [cagnulein/qdomyos-zwift#2667]
**Describe the bug**
Just trying to have my proform TDF to do automatic incline while riding hills on Zwift.

**To Reproduce**
I detect my proform and start riding, i have power, rpm and even HR, but nothing go from Zwift to the bike.

**Expected behavior**
Just hoping to see my bike following the same incline in the game.
I did a few test with many option, maybe its something i did.

**Smartphone (please complete the following information):**
[debug-Wed_Oct_16_11_16_35_2024.log](https://github.com/user-attachments/files/17398458/debug-Wed_Oct_16_11_16_35_2024.log)

 - Device: Iphone 15
 - OS: IOS 18.0.1

[debug-Wed_Oct_16_11_16_35_2024.log](https://github.com/user-attachments/files/17398467/debug-Wed_Oct_16_11_16_35_2024.log)

**Comment by mminville** [comment]
I was using Proform Studio bike, i'll retry with the one on the screenshot.
________________________________
De : Roberto Viola ***@***.***>
Envoyé : 16 octobre 2024 11:45
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)


Hi @mminville<https://github.com/mminville> did you enable this and restarted qz?

Screenshot_20241016-174341.png (view on web)<https://github.com/user-attachments/assets/b8c3cc1a-742d-4e29-88c2-bd4c27506935>

**Comment by cagnulein** [comment]
Ok basically you have to find the Right one for you. Remember to leave only
one enabled and restart qz always after a modification

**Comment by mminville** [comment]
Hi,

Yeah i tried a few one and was on this one since it was working for connection.
But with your advice, Proform TDF 1.0 it work !!! wow, i did a 5km could not stop hahaha

So i dont touch the gear on the bike either ?  What if i want to sprint, is there any way to increase the resistance or its like i'm at max like on a real bike ?


________________________________
De : Roberto Viola ***@***.***>
Envoyé : 16 octobre 2024 11:49
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)

Ok basically you have to find the Right one for you. Remember to leave only
one enabled and restart qz always after a modification

**Comment by cagnulein** [comment]
Yes use the gear tile to make a sprint like on a real bike! You can enable
it on the tiles settings  on qz

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 18:07 Martin Minville <
***@***.***> ha scritto:

**Comment by mminville** [comment]
Splendid!

Thank you so much!

________________________________
De : Roberto Viola ***@***.***>
Envoyé : 16 octobre 2024 12:19
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)

Yes use the gear tile to make a sprint like on a real bike! You can enable
it on the tiles settings on qz

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 18:07 Martin Minville <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
Please remember to join the swag bag from the app to support my work!
Thanks!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 18:26 Martin Minville <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
Thanks for the coffees!

**Comment by mminville** [comment]
Hi,

I did a good test yesterday, everything is great and challenging on hills.
I was using the phone volume to change gear.

But on the flat, its like i'm missing some small gear on this bike if i want to sprint.
Look like the bike limitation, nothing i can do.  Its max 26 and even if i spin it like hell, it stay at 315watts

Thank!
________________________________
De : Roberto Viola ***@***.***>
Envoyé : 16 octobre 2024 12:19
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)


Yes use the gear tile to make a sprint like on a real bike! You can enable
it on the tiles settings on qz

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 18:07 Martin Minville <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
hah it could be, unfortunately there is nothing that in can do for this :( you can try to add a magnet on the wheel and then you have to recalibrate the bike, but i don't know how much is the efforts

**Comment by mminville** [comment]
I got my garmin rs200 pedal and i just dont understant.... they give me like 1000 watts for a minmal effort...

I try to calibrate them but always get a 1280 error.... really disapointed by them....  I wanted something real... and its worse.

I'll try to get some support from garmin....

________________________________
De : Roberto Viola ***@***.***>
Envoyé : 16 octobre 2024 12:19
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)


Yes use the gear tile to make a sprint like on a real bike! You can enable
it on the tiles settings on qz

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 16 ott 2024 alle 18:07 Martin Minville <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
Is this happening also with their app?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 19 nov 2024 alle 23:03 Martin Minville <
***@***.***> ha scritto:

**Comment by mminville** [comment]
Yeah, same wierd number with my edge 1040.
They are replacing it, cant do anything with it.

________________________________
De : Roberto Viola ***@***.***>
Envoyé : 20 novembre 2024 01:42
À : cagnulein/qdomyos-zwift ***@***.***>
Cc : Martin Minville ***@***.***>; Mention ***@***.***>
Objet : Re: [cagnulein/qdomyos-zwift] Trying to have incline on my proform TDF [BUG] (Issue #2667)

Is this happening also with their app?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 19 nov 2024 alle 23:03 Martin Minville <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
Ok yes they seem broken

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 20 nov 2024 alle 12:58 Martin Minville <
***@***.***> ha scritto:
